### ISSUE TITLE
No font asset found for - ZTE BLADE devices

### ISSUE THREAD

**Issue by cagkanciloglu** [afollestad/material-dialogs#1275]
This crash seems to happen only in ZTE BLADE V7 LITE phone.

Fatal Exception: java.lang.IllegalArgumentException: No font asset found for 
       at com.afollestad.materialdialogs.MaterialDialog$Builder.typeface(MaterialDialog.java:690)
       at com.afollestad.materialdialogs.MaterialDialog$Builder.(MaterialDialog.java:556)
       at com.utils.DialogManager.showLoginSignUpAlert(DialogManager.java:57)
       at com.fragments.LoginFragment.onClick(LoginFragment.java:120)
       at android.view.View.performClick(View.java:5265)
       at android.view.View$PerformClick.run(View.java:21534)
       at android.os.Handler.handleCallback(Handler.java:815)
       at android.os.Handler.dispatchMessage(Handler.java:104)
       at android.os.Looper.loop(Looper.java:207)
       at android.app.ActivityThread.main(ActivityThread.java:5728)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:789)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:679)

 - Material Dialogs version: `0.9.2.3`
 - Device OS version: `6.0`
 - Device Manufacturer: `ZTE `
 - Device Name: `BLADE V7 LITE`

**Comment by afollestad** [comment]
This isn't an issue with this library.

You're passing an empty string into the Builder.

**Comment by cagkanciloglu** [comment]
Code works for every other phone except this one though

**Comment by afollestad** [comment]
@cagkanciloglu hmm, well your log here would suggest that the "medium" String parameter is null or empty:

<img width="849" alt="screen shot 2017-01-31 at 12 39 15 pm" src="https://cloud.githubusercontent.com/assets/1820165/22479160/55e3b660-e7b2-11e6-9ad4-37762ddfac87.png">

**Comment by afollestad** [comment]
The best solution would probably be to wrap your `Builder` code with a try/catch block. 

Suppressing this exception on my end would be bad because it may actually signal a user error in some cases.

**Comment by vitkidd** [comment]
Recently, I got the same crash on ZTE BLADE A910, Android 6.0.

**Comment by afollestad** [comment]
It must be an issue with ZTE's skin.

**Comment by cagkanciloglu** [comment]
This is what I don't understand about your code maybe there is a logic behind it I don't see. The function typeface throws that exception when mediumFont is null right? On the code you first call typeface then you check whether if mediumFont is null or not. Why don't you do this check before calling typeface function or inside the typeface function? 

MaterialDialog line 556
`typeface(mediumFont, regularFont);

            if (this.mediumFont == null) {
                try {
                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)
                        this.mediumFont = Typeface.create("sans-serif-medium", Typeface.NORMAL);
                    else
                        this.mediumFont = Typeface.create("sans-serif", Typeface.BOLD);
                } catch (Exception ignored) {
                }
            }
            if (this.regularFont == null) {
                try {
                    this.regularFont = Typeface.create("sans-serif", Typeface.NORMAL);
                } catch (Exception ignored) {
                }
            }`

**Comment by afollestad** [comment]
@cagkanciloglu that sets default typefaces from the system fonts.

**Comment by afollestad** [comment]
And that isn't related to whatever happens in this device's skin.

**Comment by afollestad** [comment]
Actually I guess it could be if those default typefaces don't exist, which would be unusual. I'll make a workaround.

**Comment by trr-amsiq** [comment]
Hi @afollestad,
Has this issue been fixed? I'm seeing this issue on ZTE devices also. using 0.9.4.4. [Fabric](http://crashes.to/s/a3fa00f0534)

**Comment by afollestad** [comment]
@trr-amsiq that log shows that you are passing a value to `typeface(String, String)` in the `Builder` which is not an existing font.

https://github.com/afollestad/material-dialogs/blob/master/core/src/main/java/com/afollestad/materialdialogs/MaterialDialog.java#L1404

**Comment by trr-amsiq** [comment]
I'm not calling any typeface setter

**Comment by cagkanciloglu** [comment]
default typefaces don't exist for some models @afollestad should make a workaround. 
but he kinda doesn't give a sh*t and closed the issue.
put try-catch around the block and show default dialog in catch.

**Comment by trr-amsiq** [comment]
@afollestad really??

**Comment by afollestad** [comment]
@trr-amsiq welcome to make a pull request.

**Comment by trr-amsiq** [comment]
I added these two lines to styles.xml and it stopped crashing
```
<item name="md_medium_font">NotoSans-Regular.ttf</item>
        <item name="md_regular_font">NotoSans-Regular.ttf</item>
```

**Comment by afollestad** [comment]
Should be fixed in 0.9.4.5.
